Fix public API tests

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131390
* __->__ #131386

This PR fixes a bug in `test_correct_module_names` introduced in #130497. It also addresses post-fix test failures in:
* `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
* `torch/library.py` - add `register_vmap` to `__all__`
* `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
* `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API
* `torch/distributed/pipelining/schedules.py` - add `ZeroBubbleAlgorithm` to `__all__`

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

jbschlosser committed:
Introduce sanity check lint + fix for public API tests

[ghstack-poisoned]

jbschlosser committed:
Update on "Introduce sanity check lint + fix for public API tests"


This PR:
* Introduces a new sanity check for the public API tests in `.github/scripts/check_public_API_tests.py`.
    * Validates two public API tests:
        1. Ensures `test_correct_module_names` fails when a new file adds an invalid public API function (e.g. one whose `__module__` is unset).
        2. Ensures `test_modules_can_be_imported` fails when an existing file underneath `torch/` introduces an invalid public API function.
    * Runs this on the lint shard in CI as part of the linter that validates test running.
* Fixes a pre-existing bug in `test_correct_module_names` introduced in #130497: `test_module(modname) ->` `test_module(mod)`
   * Addresses post-fix test failures in:
       * `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
       * `torch/cuda/gds.py` - introduce empty `__all__` to avoid `Storage` being re-exported as a public API
       * `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
       * `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API

[ghstack-poisoned]

jbschlosser committed:
Update on "Introduce sanity check lint + fix for public API tests"


This PR fixes a bug in `test_correct_module_names` introduced in #130497:
* `test_module(modname) ->` `test_module(mod)`
* Addresses post-fix test failures in:
    * `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
    * `torch/cuda/gds.py` - introduce empty `__all__` to avoid `Storage` being re-exported as a public API
    * `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
    * `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API

[ghstack-poisoned]

albanD reviewed (approved):
Thanks a lot for fixing this!

Suppressing BC-linter as the API was introduced by mistake recently. So we're happy to break BC for it.

jbschlosser commented:
@pytorchbot merge -i

jbschlosser commented:
@pytorchbot merge -i

mikaylagawarecki commented on torch/cuda/gds.py:
```diff
@@ -1,11 +1,14 @@
 from torch.types import Storage
 
 
+__all__: List[str] = []
```
Thank you! And...you might want to rebase as my PR that added this file got reverted :D https://github.com/pytorch/pytorch/pull/130633

jbschlosser committed:
Update on "Fix public API tests"


This PR fixes a bug in `test_correct_module_names` introduced in #130497. It also addresses post-fix test failures in:
* `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
* `torch/cuda/gds.py` - introduce empty `__all__` to avoid `Storage` being re-exported as a public API
* `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
* `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API

[ghstack-poisoned]

jbschlosser committed:
Update on "Fix public API tests"


This PR fixes a bug in `test_correct_module_names` introduced in #130497. It also addresses post-fix test failures in:
* `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
* `torch/cuda/gds.py` - introduce empty `__all__` to avoid `Storage` being re-exported as a public API
* `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
* `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API

[ghstack-poisoned]

jbschlosser committed:
Update on "Fix public API tests"


This PR fixes a bug in `test_correct_module_names` introduced in #130497. It also addresses post-fix test failures in:
* `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
* `torch/library.py` - add `register_vmap` to `__all__`
* `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
* `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API

[ghstack-poisoned]

jbschlosser committed:
Update on "Fix public API tests"


This PR fixes a bug in `test_correct_module_names` introduced in #130497. It also addresses post-fix test failures in:
* `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
* `torch/library.py` - add `register_vmap` to `__all__`
* `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
* `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API
* `torch/distributed/pipelining/schedules.py` - add `ZeroBubbleAlgorithm` to `__all__`

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge

justinchuby commented:
Can members from typing and typing_extensions be exempted? They seem very common

clee2000 commented:
@pytorchbot revert -m "reverting this to revert something else, only action you should need to do is to rebase and merge again, sorry for the churn" -c ghfirst

jbschlosser commented:
@pytorchbot rebase

jbschlosser committed:
Update on "Fix public API tests"


This PR fixes a bug in `test_correct_module_names` introduced in #130497. It also addresses post-fix test failures in:
* `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
* `torch/library.py` - add `register_vmap` to `__all__`
* `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
* `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API
* `torch/distributed/pipelining/schedules.py` - add `ZeroBubbleAlgorithm` to `__all__`

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge

jbschlosser committed:
Update on "Fix public API tests"


This PR fixes a bug in `test_correct_module_names` introduced in #130497. It also addresses post-fix test failures in:
* `torch/ao/quantization/__init__.py` - set the correct `__module__` for several public API helpers
* `torch/library.py` - add `register_vmap` to `__all__`
* `torch/nn/attention/flex_attention.py` - make `round_up_to_multiple` private by prepending an underscore
* `torch/storage.py` - introduce `__all__` to avoid `Self` being re-exported as a public API
* `torch/distributed/pipelining/schedules.py` - add `ZeroBubbleAlgorithm` to `__all__`

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge
